src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -635,19 +637,18 @@ int xdg_autostart_service_generate_unit(
                         e_not_show_in);
         }
 
-        r = xdg_autostart_generate_desktop_condition(f,
+        r = xdg_autostart_generate_desktop_condition(service, f,
                                                      "gnome-systemd-autostart-condition",
                                                      service->autostart_condition);
         if (r < 0)
                 return r;
 
-        r = xdg_autostart_generate_desktop_condition(f,
+        r = xdg_autostart_generate_desktop_condition(service, f,
                                                      "kde-systemd-start-condition",
                                                      service->kde_autostart_condition);
         if (r < 0)
                 return r;
 
-        (void) generator_add_symlink(dest, "xdg-desktop-autostart.target", "wants", service->name);
-
-        return 0;
+        log_debug("%s: symlinking %s in xdg-desktop-autostart.target/.wants…", service->path, service->name);
+        return generator_add_symlink(dest, "xdg-desktop-autostart.target", "wants", service->name);
```

Comment:
This is more than a debug log change - it couldn't fail before, but now it can. Was it intended? If so, it probably deserves a mention in the commit message?

Reply:
The return value is ignored in the caller. I'll add a mention in the commit message.